src/GafferScene/Set.cpp

```diff
@@ -192,75 +194,105 @@ IECore::ConstInternedStringVectorDataPtr Set::computeSetNames( const Gaffer::Con
 {
 	ConstInternedStringVectorDataPtr inNamesData = inPlug()->setNamesPlug()->getValue();
 
-	InternedString name = namePlug()->getValue();
-	if( !name.string().size() )
+	const std::string &names = namePlug()->getValue();
+	if( !names.size() )
 	{
 		return inNamesData;
 	}
 
-	const std::vector<InternedString> &inNames = inNamesData->readable();
-	if( std::find( inNames.begin(), inNames.end(), name ) != inNames.end() )
+	if( modePlug()->getValue() == Remove )
 	{
 		return inNamesData;
 	}
 
-	if( modePlug()->getValue() == Remove )
+	InternedStringVectorDataPtr result = NULL;
+
+	Tokenizer namesTokenizer( names, boost::char_separator<char>( " " ) );
```

Comment:
The boost tokenizer is really pretty poor performance-wise. The `tokenize` method in Gaffer/StringAlgo.h can tokenize straight into a `vector<InternedString>`, and in most cases avoids any string allocations at all because the InternedStrings already exist. I suggest we use that here too - you can see an example in the `compute()` method in this file.